cmd/juju/metricsdebug/collectmetrics.go

```diff
@@ -163,7 +163,7 @@ func (c *collectMetricsCommand) Run(ctx *cmd.Context) error {
 	resultChannel := make(chan string, len(runResults))
 	for _, result := range runResults {
 		r := result
-		if r.Error != nil {
+		if err := r.Error; err != nil {
 			fmt.Fprintf(ctx.Stdout, "failed to collect metrics: %v\n", err)
```

Comment:
The general assumption is that anything written to STDOUT is meant to be consumed by users/tools so it's a good practice  to dump errors to STDERR.

```suggestion
			_, _ = fmt.Fprintf(ctx.Stderr, "failed to collect metrics: %v\n", err)
```

Reply:
Sounds like something for a separate PR. WDYT?

Reply:
In my view it does fit the context of this set of changes given that sending the error message to the appropriate sink is definitely an improvement.

Reply:
I can do that, of course.  But it changes the scope of this PR significantly from "Stop ignoring one specific error value" to "send a large number of errors to STDERR instead of STDOUT".